### PR TITLE
Coverity and SCA fixes

### PR DESCRIPTION
**Pull request by dalgaaf**

### CODE REVIEW

File: src/osd/OSD.cc [hunk 1]
```diff
@@ -8871,7 +8871,7 @@ void OSD::handle_pg_query(OpRequestRef op)
       ConnectionRef con = service.get_con_osd_cluster(from, osdmap->get_epoch());
       if (con) {
 	MOSDPGLog *mlog = new MOSDPGLog(
-	  it->second.from, it->second.to,
+	  it->second.to, it->second.from,
```
Comment (tchaikov) [inline src/osd/OSD.cc]: this looks like a bug fix!

File: src/osd/OSD.cc [hunk 2]
```diff
@@ -8881,9 +8881,9 @@ void OSD::handle_pg_query(OpRequestRef op)
       notify_list[from].push_back(
 	make_pair(
 	  pg_notify_t(
-	    it->second.from, it->second.to,
-	    it->second.epoch_sent,
+	    it->second.to, it->second.from,
```
Comment (tchaikov) [inline src/osd/OSD.cc]: and this!

File: src/tools/rbd/Utils.cc [hunk 3]
```diff
@@ -633,7 +633,7 @@ int validate_snapshot_name(at::ArgumentModifier mod,
 int get_image_options(const boost::program_options::variables_map &vm,
 		      bool get_format, librbd::ImageOptions *opts) {
   uint64_t order = 0, stripe_unit = 0, stripe_count = 0, object_size = 0;
-  uint64_t features = 0, features_clear = 0, features_set = 0;
+  uint64_t features = 0, features_clear = 0;
```
Comment (tchaikov) [inline src/tools/rbd/Utils.cc]: @dillaman could you take a look at this commit?

File: src/tools/rbd/Utils.cc [hunk 4]
```diff
@@ -672,6 +672,7 @@ int get_image_options(const boost::program_options::variables_map &vm,
   if (vm.count(at::IMAGE_SHARED) && vm[at::IMAGE_SHARED].as<bool>()) {
     if (features_specified) {
       features &= ~RBD_FEATURES_SINGLE_CLIENT;
+      features_set_specified = true;
```
Comment (dillaman) [inline src/tools/rbd/Utils.cc]: Just yank the ```features_set_specified``` variable and related logic since it isn't used.

**Comment by tchaikov** [comment]
```
  3/178 Test   #5: cephtool-test-mon.sh ....................***Failed   58.25 sec
...
```

File: src/crush/CrushCompiler.cc [hunk 5]
```diff
@@ -1016,6 +1016,11 @@ int CrushCompiler::parse_choose_args(iter_t const& i)
   }
   crush_choose_arg_map arg_map;
   arg_map.size = crush.get_max_buckets();
+  if (arg_map.size < 0) {
```
Comment (tchaikov) [inline src/crush/CrushCompiler.cc]:
```
/var/ceph/ceph/src/crush/CrushCompiler.cc:1019:20: warning: comparison of unsigned expression < 0 is always false [-Wtype-limits]
   if (arg_map.size < 0) {
       ~~~~~~~~~~~~~^~~
```

**Comment by tchaikov** [comment]
@dalgaaf  ping?

**Comment by tchaikov** [comment]
@dalgaaf ping?

**Comment by liewegas** [comment]
needs rebase

**Comment by liewegas** [comment]
@amitkumar50 care to adopt and rebase this set of fixes?

**Comment by tchaikov** [comment]
superseded by #17431

File: src/client/Client.cc [hunk 6]
```diff
@@ -1730,7 +1730,6 @@ int Client::make_request(MetaRequest *request,
     r = request->get_abort_code();
     request->item.remove_myself();
     unregister_request(request);
-    put_request(request); // ours
```
Comment (ukernel) [inline src/client/Client.cc]: this one introduced http://tracker.ceph.com/issues/21274

**Pull request closed without merging**